Gems/ROS2/Code/Source/RobotImporter/FixURDF/FixURDF.cpp

```diff
@@ -0,0 +1,145 @@
+            inertial->append_node(mass);
+            xml_node<>* inertia = urdf->document()->allocate_node(node_element, "inertia");
+
+            inertia->append_attribute(urdf->document()->allocate_attribute("ixx", "1."));
+            inertia->append_attribute(urdf->document()->allocate_attribute("ixy", "0."));
+            inertia->append_attribute(urdf->document()->allocate_attribute("ixz", "0."));
+
+            inertia->append_attribute(urdf->document()->allocate_attribute("iyx", "0."));
+            inertia->append_attribute(urdf->document()->allocate_attribute("iyy", "1."));
+            inertia->append_attribute(urdf->document()->allocate_attribute("iyz", "0."));
+
+            inertia->append_attribute(urdf->document()->allocate_attribute("izx", "0."));
+            inertia->append_attribute(urdf->document()->allocate_attribute("izy", "0."));
+            inertia->append_attribute(urdf->document()->allocate_attribute("izz", "1."));
+
+            inertial->append_node(inertia);
+
+            xml_node<>* origin = urdf->document()->allocate_node(node_element, "origin");
+            origin->append_attribute(urdf->document()->allocate_attribute("xyz", "0. 0. 0."));
+            inertial->append_node(origin);
+
+            auto* name = link->first_attribute("name");
+            if (name)
+            {
+                modifiedLinks.push_back(name->value());
+            }
+            link->append_node(inertial);
+        }
+
+        return modifiedLinks;
+    }
+
+    //! Handles a case of multiple joints and the link sharing a common names which causes SDF error2 (but is fine in URDF)
+    //! Function will add a suffix "_dup" to the name of the joint if it is also the name of a link.
+    //! If there are name collisions in links, this will not be able to fix it, the SDF parser will throw an error.
+    //! @param urdf URDF to modify.
+    //! @returns a list of links that were modified
+    AZStd::vector<AZStd::string> RenameDuplicatedJoints(AZ::rapidxml::xml_node<>* urdf)
+    {
+        using namespace AZ::rapidxml;
+        AZStd::vector<AZStd::string> modifiedLinks;
+        AZStd::unordered_map<AZStd::string, unsigned int> linkAndJointsName;
+        for (xml_node<>* link = urdf->first_node("link"); link; link = link->next_sibling("link"))
+        {
+            auto* name = link->first_attribute("name");
+            if (name)
+            {
+                linkAndJointsName.insert(AZStd::make_pair(name->value(), 0));
+            }
+        }
+        for (xml_node<>* joint = urdf->first_node("joint"); joint; joint = joint->next_sibling("joint"))
+        {
+            auto* name = joint->first_attribute("name");
+            if (name)
+            {
+                if (linkAndJointsName.contains(name->value()))
+                {
+                    unsigned int& count = linkAndJointsName[name->value()];
+                    auto newName = AZStd::string::format("%s_dup%d", name->value(), count);
```

Comment:
This should be %u for an unsigned int.

```suggestion
                    auto newName = AZStd::string::format("%s_dup%u", name->value(), count);
```